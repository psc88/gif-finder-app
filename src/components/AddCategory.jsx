import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');
    const onInputChanged = ({ target }) => {
        setInputValue(target.value);
    }
    const onSubmit = (e) => {
        e.preventDefault();
        const newValue = inputValue.trim();
        if (newValue.length <= 3) return;
        onNewCategory(newValue)
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar Gif"
                value={inputValue}
                onChange={onInputChanged}
            />
        </form>
    )
}
