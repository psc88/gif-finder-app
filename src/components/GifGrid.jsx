import { useEffect, useState } from "react";
import { getGif } from "../helpers/getGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect( () => {
        getImages(category);
    }, []);

    const getImages = async () => {
        const newImages = await getGif( category );
        setImages(newImages);
    }
    
    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
                {
                    images.map( ( image ) => (
                        <GifGridItem 
                            key={ image.id } 
                            { ...image }
                        />
                    ))
                }
            </div>
        </>
    )
}