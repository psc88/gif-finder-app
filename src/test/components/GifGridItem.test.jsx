import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en GifGridItem', () => {
    
    const title = 'Saitama';
    const url   = 'https://one-punch.com/'
    
    test('Tiene que hacer match con el snapshot', () => { 
        const { container } = render(<GifGridItem title={title} url={url}/>)
        expect( container ).toMatchSnapshot();
     });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render( <GifGridItem title={title} url={url}/> );
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect( alt ).toBe( title );
        expect( src ).toBe( url );
     });
     
     test('debe de mostar el titulo en el componente', () => { 
        render( <GifGridItem title={title} url={url}/> );
        expect( screen.getByText( title ) ).toBeTruthy();
      })
 })