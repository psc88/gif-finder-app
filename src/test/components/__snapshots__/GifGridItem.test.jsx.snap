// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Tiene que hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
